pkg/resources/status/phase.go

```diff
@@ -99,3 +99,59 @@ func RegisterPhaseWithOrigCluster(
 
 	return nil
 }
+
+// RegisterStatusWithOrigCluster updates the status to reflect that in `modifiedCluster`
+// and adjusts the conditions according to whether the Phase is healthy
+func RegisterStatusWithOrigCluster(
+	ctx context.Context,
```

Comment:
Massive comment here.
I have created a new function because there are several places in the code that rely on
`func RegisterPhaseWithOrigCluster(`
That function now applies only the Phase and PhaseReason, and discards any other information provided via the `modifiedCluster` argument.
Given the comment on that function, not sure this was intended.
If not, then we should fix that one and discard this one.